guardian/cdk
# Prevent `yarn.lock` from being committed to the repo

## What does this change?

This repository currently uses `npm` over `yarn` due to some complications in releasing the library with `yarn`. Now that we've settled on using `npm`, we should make it difficult and/or impossible to commit `yarn.lock` files accidentally.

This PR adds a script that is run in `script/build` (the first step in CI) to check for a `yarn.lock` file and fail if found.

## Does this change require changes to existing projects or CDK CLI?

Don't think so, no.

## How to test

In the root of the repo, to see it fail: 

```
$ touch yarn.lock
$ ./script/build # see script fail

$ rm yarn.lock
$ ./script/build # see script pass
```

## How can we measure success?

We will never see `yarn.lock` in this repo! Until we later need to go back to yarn!!
